pkg/operator/csidriveroperator/olmremovalcontroller.go

```diff
@@ -0,0 +1,463 @@
+			return c.markFinished("CSI driver installed by OLM is not preset")
+		}
+	}
+
+	// 3. Wait until OLM removes the the operator deployment
+	removed, err := c.ensureOperatorDeploymentRemoved(c.olmOperatorNamespace, c.olmOptions.OLMOperatorDeploymentName)
+	if err != nil {
+		return err
+	}
+	if !removed {
+		klog.V(4).Infof("OLMOperatorRemovalController.Sync waiting for OLM to delete the operator")
+		return c.markProgressing("Waiting for OLM to delete the operator")
+	}
+
+	// 4. Delete the driver Deployment. Note that the old operator pods can be still running at this point.
+	removed, err = c.ensureDriverDeploymentRemoved(ctx, c.olmOptions.CSIDriverNamespace, c.olmOptions.CSIDriverDeploymentName)
+	if err != nil {
+		return err
+	}
+	if !removed {
+		klog.V(4).Infof("CSIDriverMigrationController.Sync waiting for the driver Deployment to disappear")
+		return c.markProgressing("Waiting for OLM driver Deployment to be deleted")
+	}
+
+	// 5. Delete the driver DaemonSet.
+	removed, err = c.ensureDriverDaemonSetRemoved(ctx, c.olmOptions.CSIDriverNamespace, c.olmOptions.CSIDriverDaemonSetName)
+	if err != nil {
+		return err
+	}
+	if !removed {
+		klog.V(4).Infof("CSIDriverMigrationController.Sync waiting for the driver DaemonSet to disappear")
+		return c.markProgressing("Waiting for OLM driver DaemonSet to be deleted")
+	}
+
+	// 6. Remove CR
+	removed, err = c.ensureCRRemoved(ctx, c.olmOptions.CRResource)
+	if err != nil {
+		return err
+	}
+	if !removed {
+		klog.V(4).Infof("OLMOperatorRemovalController.Sync waiting for the CR to disappear")
+		return c.markProgressing("Waiting for the old operator CR to be deleted")
+	}
+
+	klog.V(4).Infof("OLMOperatorRemovalController.Sync done!")
+	return c.markFinished("CSI driver has been removed from OLM")
+}
+
+func (c *OLMOperatorRemovalController) findSubscription(ctx context.Context) (string, string, string, bool, error) {
+	subscriptions, err := c.dynamicClient.Resource(subscriptionResourceGVR).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return "", "", "", false, err
+	}
+	for _, obj := range subscriptions.Items {
+		pkg, found, err := unstructured.NestedString(obj.Object, "spec", "name")
+		if !found {
+			continue
+		}
+		if err != nil {
+			return "", "", "", false, err
+		}
+		source, found, err := unstructured.NestedString(obj.Object, "spec", "source")
+		if !found {
+			continue
+		}
+		if err != nil {
+			return "", "", "", false, err
+		}
+		sourceNamespace, found, err := unstructured.NestedString(obj.Object, "spec", "sourceNamespace")
+		if !found {
+			continue
+		}
+		if err != nil {
+			return "", "", "", false, err
+		}
+
+		if pkg == c.olmOptions.OLMPackageName && source == olmSource && sourceNamespace == olmSourceNamespace {
+			csvName, _, err := unstructured.NestedString(obj.Object, "status", "currentCSV")
+			if err != nil {
+				return "", "", "", false, err
+			}
+			klog.V(2).Infof("Found subscription %s/%s with CSV %s", obj.GetNamespace(), obj.GetName(), csvName)
```

Comment:
V(2) might be too noisy

Reply:
Moved to v(4), however, it should be printed just once in ideal case.